Malformed plugin name causes agent to panic

When trying to reference a plugin that can't be resolved. In this case it was a private git repo address:

```
github.com/git@github.com:buildkite-plugins/shellcheck-buildkite-plugin.git-buildkite-plugin
```

```
panic: runtime error: invalid memory address or nil pointer dereference
--
  | [signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xb03c11]
  |  
  | goroutine 1 [running]:
  | github.com/buildkite/agent/bootstrap.(*Bootstrap).executePluginHook(0xc42013d860, 0xd6eab5, 0x8, 0x0, 0x0)
  | /go/src/github.com/buildkite/agent/bootstrap/bootstrap.go:410 +0x91
  | github.com/buildkite/agent/bootstrap.(*Bootstrap).tearDown(0xc42013d860, 0xd9fb38, 0xc4201df8b8)
  | /go/src/github.com/buildkite/agent/bootstrap/bootstrap.go:352 +0xb0
  | github.com/buildkite/agent/bootstrap.(*Bootstrap).Start.func1(0xc42013d860)
  | /go/src/github.com/buildkite/agent/bootstrap/bootstrap.go:61 +0x2f
  | github.com/buildkite/agent/bootstrap.(*Bootstrap).Start(0xc42013d860, 0x80)
  | /go/src/github.com/buildkite/agent/bootstrap/bootstrap.go:97 +0x49b
  | github.com/buildkite/agent/clicommand.glob..func6(0xc4200949a0)
  | /go/src/github.com/buildkite/agent/clicommand/bootstrap.go:272 +0x4d4
  | github.com/buildkite/agent/vendor/github.com/urfave/cli.HandleAction(0xbb2660, 0xd9fb98, 0xc4200949a0, 0xc420054f00, 0x0)
  | /go/src/github.com/buildkite/agent/vendor/github.com/urfave/cli/app.go:504 +0x7c
  | github.com/buildkite/agent/vendor/github.com/urfave/cli.Command.Run(0xd6f366, 0x9, 0x0, 0x0, 0x0, 0x0, 0x0, 0xd7f964, 0x1b, 0x0, ...)
  | /go/src/github.com/buildkite/agent/vendor/github.com/urfave/cli/command.go:210 +0xaad
  | github.com/buildkite/agent/vendor/github.com/urfave/cli.(*App).Run(0xc4201a61c0, 0xc42000a060, 0x2, 0x2, 0x0, 0x0)
  | /go/src/github.com/buildkite/agent/vendor/github.com/urfave/cli/app.go:259 +0x740
  | main.main()
  | /go/src/github.com/buildkite/agent/main.go:102 +0x52e
```

--- a/bootstrap/bootstrap.go
+++ b/bootstrap/bootstrap.go
@@ -389,15 +389,15 @@ func (b *Bootstrap) PluginPhase() error {
 		return errors.Wrap(err, "Failed to parse plugin definition")
 	}
 
-	b.plugins = make([]*pluginCheckout, len(plugins))
+	b.plugins = []*pluginCheckout{}
 
-	for idx, p := range plugins {
+	for _, p := range plugins {
 		checkout, err := b.checkoutPlugin(p)
 		if err != nil {
 			return errors.Wrapf(err, "Failed to checkout plugin %s", p.Name())
 
 		}
-		b.plugins[idx] = checkout
+		b.plugins = append(b.plugins, checkout)
 	}
 
 	// Now we can run plugin environment hooks too

--- a/bootstrap/integration/plugin_integration_test.go
+++ b/bootstrap/integration/plugin_integration_test.go
@@ -73,6 +73,26 @@ func TestRunningPlugins(t *testing.T) {
 	tester.RunAndCheck(t, env...)
 }
 
+func TestMalformedPluginNamesDontCrashBootstrap(t *testing.T) {
+	t.Parallel()
+
+	tester, err := NewBootstrapTester()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tester.Close()
+
+	env := []string{
+		`BUILDKITE_PLUGINS=["sdgmdgn.@$!sdf,asdf#llamas"]`,
+	}
+
+	if err = tester.Run(t, env...); err == nil {
+		t.Fatal("Expected the bootstrap to fail")
+	}
+
+	tester.CheckMocks(t)
+}
+
 type testPlugin struct {
 	*gitRepository
 }

